### DISCUSSION TITLE
Migration steps from Snap to docker AIO?

### DISCUSSION THREAD

**Discussion by wouterverduin** [nextcloud/all-in-one#3349] [Questions] [answered]
Hi all!

I am running nextcloud for years now and very content with the software. However, my server needs an update and figured this was the moment to switch from the snap i am using to the docker AIO considering that better feeds my expertise and server goals.

However, i cant really seem to find the right guides and pieced together what i think would be the best steps. The reason for my post is before i start and destroy everything i would hope for feedback if i am missing something or making stupid steps?
Otherwise this might be used for future references for others embarking on a similar project since piecing this together took some time.

The steps i figured so far:

**Step 1:**
On my running nextcloud snap installed the user migration tool. In my case i have one user i want to maintain and export. Use the tool to export everything for later import.

**Step 2:**
Data direction is on a Freenas drive mounted as CIFS. Mounted this point on my new ubuntu install with docker under the folder "/NEXTCLOUD". Directly in this folder are my "username_folder" , files_external folder, index.html and some logs.

**Step 3:**
Prepare the docker command. So far i am aiming to run this:

`sudo docker run \
--init \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8080:8080 \
--env APACHE_PORT=11000 \
--env APACHE_IP_BINDING=0.0.0.0 \
--volume /root/Nextcloud/config:/mnt/docker-aio-config \
-e NEXTCLOUD_DATADIR="/NEXTCLOUD" \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
`

I am running a caddy reverse proxy so thats why i am aiming to run this config. The place i want my docker files for possible future migration ease is bound to '/root/Nextcloud/config'.

**Step 4:**
Adjusting Caddy to new situation using the examples on github: https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

**Step 5:**
Shutdown the snap. Run the docker command above. Head to my reverse proxy adress. Register and imigrate my existing user and hopefully see all my files there.

Am i missing steps? Will this work? Any advice is very welcome. If everything worked i will write this up with more explicit steps for now i hope to get suggestions before screwing things up:)

Thanks in advance!

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one/blob/main/migration.md
